src/plone/restapi/tests/test_field_serializer.py

```diff
@@ -0,0 +1,90 @@
+# -*- coding: utf-8 -*-
+from datetime import datetime
+from plone.app.contenttypes.behaviors.richtext import IRichText
+from plone.dexterity.utils import createContentInContainer
+from plone.dexterity.utils import iterSchemata
+from plone.restapi.dxfields import DateFieldSerializer
+from plone.restapi.dxfields import DateTimeFieldSerializer
+from plone.restapi.dxfields import DefaultFieldSerializer
+from plone.restapi.interfaces import IFieldSerializer
+from plone.restapi.testing import PLONE_RESTAPI_FUNCTIONAL_TESTING
+from unittest2 import TestCase
+from zope.component import getMultiAdapter
+from zope.interface.verify import verifyClass
+from zope.schema import getFields
+import json
+
+
+class TestDexterityFieldSerializing(TestCase):
+    layer = PLONE_RESTAPI_FUNCTIONAL_TESTING
+
+    def setUp(self):
+        self.portal = self.layer['portal']
+        self.request = self.layer['request']
+
+    def test_boolean(self):
+        folder = createContentInContainer(self.portal, u'Folder',
+                                          exclude_from_nav=True)
+        value = self.serialize(folder, u'exclude_from_nav')
+        self.assertEquals(True, value)
+        self.assertEquals(value, json.loads(json.dumps(value)))
+
+    def test_textline(self):
+        folder = createContentInContainer(self.portal, u'Folder',
+                                          title=u'My Folder')
+        value = self.serialize(folder, u'title')
+        self.assertEquals(u'My Folder', value)
+        self.assertEquals(value, json.loads(json.dumps(value)))
+
+    def test_choice(self):
+        folder = createContentInContainer(self.portal, u'Folder',
+                                          language=u'lo')
+        value = self.serialize(folder, u'language')
+        self.assertEquals(u'lo', value)
+        self.assertEquals(value, json.loads(json.dumps(value)))
+
+    def test_date(self):
+        folder = createContentInContainer(
+            self.portal, u'Folder',
+            effective_date=datetime(2017, 10, 22, 19, 45, 1))
+        value = self.serialize(folder, u'effective')
+        self.assertEquals(u'2017-10-22T19:45:01', value)
+        self.assertEquals(value, json.loads(json.dumps(value)))
+
+    def test_richtext(self):
+        document = createContentInContainer(
+            self.portal, u'Document',
+            text=IRichText['text'].fromUnicode(u'<p>Some Text</p>'))
+        value = self.serialize(document, u'text')
+        self.assertEquals({'raw': u'<p>Some Text</p>',
+                           'mimeType': 'text/html',
+                           'outputMimeType': 'text/x-html-safe',
+                           'encoding': 'utf-8'}, value)
```

Comment:
How should richtext field be represented in the JSON output?

I see three possibilites:

1. Provide all infos of the richtext object, as above, so that the client knows which mimetype / encoding the value has.
2. Just return the raw output, so that we do not have a nested dict but a string value as with textline fields (e.g. `title`)
3. Same as 2 but convert the string to the `outputMimeType` (transforms).

What do you think is best?